pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionRuntimeManager.java

```diff
@@ -417,7 +418,13 @@ public synchronized void restartFunctionInstances(String tenant, String namespac
                             .type(MediaType.APPLICATION_JSON)
                             .entity(new ErrorData(fullFunctionName + " has not been assigned yet")).build());
                 }
-                this.functionAdmin.functions().restartFunction(tenant, namespace, functionName);
+                if (ComponentType.SOURCE == componentType) {
```

Comment:
following above comment, so we can have the `componentType` as `assignment .getInstance().getFunctionMetaData().getFunctionDetails().getComponentType()`.